Lib/unittest/test/testmock/testasync.py

```diff
@@ -528,11 +528,158 @@ class AsyncMockAssert(unittest.TestCase):
+
+        with self.assertRaises(AssertionError):
+            self.mock.assert_awaited()
+        with self.assertRaises(AssertionError):
+            self.mock.assert_called()
+
+    def test_assert_has_calls_not_awaits(self):
+        kalls = [call('NormalFoo')]
+        with self.assertWarns(RuntimeWarning):
+            self.mock('NormalFoo')
+        self.mock.assert_has_calls(kalls)
+        with self.assertRaises(AssertionError):
+            self.mock.assert_has_awaits(kalls)
+
+    def test_assert_has_mock_calls_on_async_mock(self):
+        mock = AsyncMock()
+        with self.assertWarns(RuntimeWarning):
+            mock()
+        kalls_empty = [('', (), {})]
+        self.assertEqual(mock.mock_calls, kalls_empty)
+
+        kalls_args = ([call('NormalFoo'), call('baz')])
+        with self.assertWarns(RuntimeWarning):
+            self.mock('NormalFoo')
+            self.mock('baz')
+        self.assertEqual(self.mock.mock_calls, kalls_args)
+
+        a_mock = AsyncMock(AsyncClass)
+        with self.assertWarns(RuntimeWarning):
+            a_mock.async_method()
+        self.assertEqual(a_mock.async_method.mock_calls, kalls_empty)
+        self.assertEqual(a_mock.mock_calls, [call.async_method()])
+
+        kalls_args_kwargs = [call(), call(1, 2, 3, a=4, b=5)]
+        with self.assertWarns(RuntimeWarning):
+            a_mock.async_method(1, 2, 3, a=4, b=5)
+        self.assertEqual(a_mock.async_method.mock_calls, kalls_args_kwargs)
+
+    def test_async_method_calls_recorded(self):
+        with self.assertWarns(RuntimeWarning):
+            self.mock.something(3, fish=None)
+            self.mock.something_else.something(6, cake=sentinel.Cake)
+
+        self.assertEqual(self.mock.something_else.method_calls,
+                        [("something", (6,), {'cake': sentinel.Cake})],
+                        "method calls not recorded correctly")
+        self.assertEqual(self.mock.method_calls, [
+            ("something", (3,), {'fish': None}),
+            ("something_else.something", (6,), {'cake': sentinel.Cake})
+        ],
+            "method calls not recorded correctly")
+
+    def test_async_arg_lists(self):
+        def assert_attrs(mock):
+            names = 'call_args_list', 'method_calls', 'mock_calls'
+            for name in names:
+                attr = getattr(mock, name)
+                self.assertIsInstance(attr, _CallList)
+                self.assertIsInstance(attr, list)
+                self.assertEqual(attr, [])
+
+            assert_attrs(self.mock)
+
+            with self.assertWarns(RuntimeWarning):
+                self.mock()
+                self.mock(1, 2)
+                self.mock(a=3)
+
+                self.mock.reset_mock()
+                assert_attrs(self.mock)
+
+                a_mock = AsyncMock(AsyncClass)
+                a_mock.async_method()
+                a_mock.async_method(1, a=3)
+
+                a_mock.reset_mock()
+                assert_attrs(a_mock)
```

Comment:
I think the second half should be dedented.
AFAICS, this is just a test that defines a function that is never called.